master/internal/api_experiment.go

```diff
@@ -290,23 +321,100 @@ func (a *apiServer) GetExperiments(
 	default:
 		orderExpr = fmt.Sprintf("id %s", sortByMap[req.OrderBy])
 	}
+	query = query.OrderExpr(orderExpr)
 
-	resp := &apiv1.GetExperimentsResponse{}
-	return resp, a.m.db.QueryProtof(
-		"get_experiments",
-		[]interface{}{orderExpr},
-		resp,
-		stateFilterExpr,
-		archivedExpr,
-		userFilterExpr,
-		userIDFilterExpr,
-		labelFilterExpr,
-		req.Description,
-		req.Name,
-		req.ProjectId,
-		req.Offset,
-		req.Limit,
-	)
+	// Filtering
+	if req.Description != "" {
+		query = query.Where("e.config->>'description' ILIKE ('%%' || ? || '%%')", req.Description)
+	}
+	if req.Name != "" {
+		query = query.Where("e.config->>'name' ILIKE ('%%' || ? || '%%')", req.Name)
+	}
+	if len(req.Labels) > 0 {
+		// In the event labels were removed, if all were removed we insert null,
+		// which previously broke this query.
+		query = query.Where(`string_to_array(?, ',') <@ ARRAY(SELECT jsonb_array_elements_text(
+				CASE WHEN e.config->'labels'::text = 'null'
+				THEN NULL
+				ELSE e.config->'labels' END
+			))`, strings.Join(req.Labels, ",")) // Trying bun.In doesn't work.
```

Comment:
what was up with bun.In?

Reply:
Was hoping you could do something like ```? <@ ARRAY``` but I needed to have this strings.Join instead